test/utils/ui_interaction.dart

```diff
@@ -0,0 +1,121 @@
+// Flutter imports:
+import 'package:flutter/material.dart';
+// Package imports:
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:image/image.dart' as img;
+// Project imports:
+import 'package:paintroid/app.dart';
+import 'package:paintroid/core/providers/state/canvas_state_provider.dart';
+import 'package:paintroid/core/providers/state/tools/toolbox/toolbox_state_provider.dart';
+
+import 'canvas_positions.dart';
+import 'widget_finder.dart';
+
+class UIInteraction {
+  static final UIInteraction _instance = UIInteraction._internal();
```

Comment:
Done- made them all static. Also, other (older) tests use different interactions helper functions, maybe it makes sense to refactor them all to use the same unified one. I suggest creating a new ticket for this since it will exceed the scope of this ticket. What do you thing?